src/wasm/extract.cpp

```diff
@@ -48,13 +49,15 @@ file_output::file_output(const fs::path& dir, const processed_file* f, bool writ
   }
 }
 
-bool file_output::write(const char* data, size_t n) {
-  int r;
-  if(!zip_open_) {
-		r = zip_.addFile(path_.c_str(), data, n, &zipindex_);
-		zip_open_ = true;
+bool file_output::write(char* data, size_t n) {
+  ZIPentry *z;
+  if(!file_open_) {
+    printf("Unpacking file %s\n", path_.c_str());
+    ze_ = zs_entrybegin(zip_, path_.c_str(), time(0), ZS_STORE, 0);
+    z = zs_entrydata(zip_, ze_, (uint8_t*) data, n, 0);
```

Comment:
You'd probably want to communicate that the data reintepretation is taking place here.
```suggestion
    z = zs_entrydata(zip_, ze_, reinterpret_cast<uint8_t*>(data), n, 0);
```